### DISCUSSION TITLE
limiting specific technology builds

### DISCUSSION THREAD

**Discussion by ssattler** [NREL/ReEDS-2.0#159] [Q&A: Analysis]
I would like to prohibit new builds of a specific technology (i.e. coal with CCS) after a specific date. What is the best way to do this?

**Comment by wesleyjcole** [comment]
This depends a bit on the specifics.  Do you want to retire plants when new ones are built?  Then you can use forced_retirements.csv.  If you want to just disallow new builds after a given date, you can simply increase their costs by a large amount after that date, or you can go into the model code and adjust valinv(i,v,r,t) to disallow new investment for that technology type.

**Reply by ssattler** [reply]
Where are the costs for the coal with CCS conversions? I see the /inputs/plant_characteristics/conv_ATB_2022.csv file with coal-CCS_mod, is this where you would increase the cap cost? Or is this the cost projections for new CCS builds?

**Comment by ssattler** [comment]
I think I answered my own question - it's the /inputs/capacitydata/upgrade_costs_ccs_coal.csv file, right?

**Comment by maxxb77** [comment]
Hi there - you'll find that, at least in the MB_IRA branch which I believe you're using, that the coal-ccs and gas-ccs retrofit costs and performance estimates are unit- or vintage-bin- specific. The relevant code is in writehintage.py which collects the following columns on the right from the EIA unit database and renames them to the values on the right:

T_CCSROV -> CCS_Retro_OvernightCost
T_CCSF -> CCS_Retro_FOM
T_CCSV -> CCS_Retro_VOM
T_CCSHR -> CCS_Retro_HR
T_CCSCAPA -> CCS_Retro_CapAdjust
T_CCSLOC -> CCS_Retro_LocFactor

These are then assigned to each 'v' bin via the capacity-weighted average later in the script

**Reply by ssattler** [reply]
Thanks. I really want to specify a certain amount of coal with CCS and not allow the model to build any additional beyond that amount. So I guess the better way to do this is to use valinv(i,v,r,t) as Wesley suggests above? I was able to force the amount I wanted into the ReEDS_generator_database, but how do you not allow for any additional investment of this tech?

**Comment by maxxb77** [comment]
If I'm understanding you correctly, you could do something like:

scalar max_ccs_cap '--MW-- maximum amount of CCS capacity to be available' /5000/ ; 
scalar first_year_ccs_cap 'first year for imposing the CCS capacity limit' /2028/ ;

equation eq_ccs_caplimit(t) '--MW-- limit on national CCS capacity' ;

eq_ccs_caplimit(t)$[tmodel(t)$(yeart(t)>=first_year_ccs_cap)]..

* maximum amount of CCS capacity allowed
    max_ccs_cap

    =g=

* total endogenous CCS capacity
   sum{(i,v,r)$[ccs(i)$coal(i)$valcap(i,v,r,t)], CAP(i,v,r,t) }
;


(not checked for syntax)
